src/core/datastore/syncstore.spec.js

```diff
@@ -842,6 +841,42 @@ describe('SyncStore', () => {
           expect(entities).toEqual([entity1, entity2]);
         });
     });
+
+    it('should save the entities from the backend to the cache from multiple pulls', () => {
```

Comment:
Did this test fail before the fix? Didn't the problem come from the second request finishing before the first? I'm not sure this would happen so with the mocked requests. Perhaps a delay would help and we might as well add a few more entities - 3, 5...

Also shouldn't this be added to the CacheStore suite as well? I think there are more problems there, stemming from the silent "push()". As pointed out in that recent issue.